upickle/test/src/upickle/ClassDefs.scala

```diff
@@ -55,7 +55,7 @@ object ADTs {
 object Hierarchy {
   sealed trait A
   object A{
-    implicit def rw: upickle.default.ReadWriter[A] = RW.merge(B.rw, C.rw)
+    implicit def rw: upickle.default.ReadWriter[A] = RW.merge(core.Annotator.defaultTagKey, B.rw, C.rw)
```

Comment:
For backwards compatibility, could we call the new method `RW.mergeKeyed` or something? 

I think it's reasonable to leave the existing functionality with the default in place rather than forcing people to explicitly pass in a `defaultTagKey` every time they call `merge`, and because of the use of varargs we can't give `defaultTagKey` a default value. So having a separate method `merge` vs `mergeKeyed` seems like the way to go

Reply:
Definitely, updated